functions/check_deploy.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+const http = require("http");
+const url = require("url");
+
+exports.handler = async function (event) {
+    console.log("Received check_deploy request at " + new Date() + " with path " + event.path);
+    var branch = /check_deploy\/(.*)/.exec(event.path)[1];
+
+    // Reject the request when:
+    // 1. Not a GET request;
+    // 2. Doesn’t provide required values
+    if (event.httpMethod !== "GET" || !branch ||
+        !process.env.ACCESS_TOKEN || !process.env.WORDLES_REPO_OWNER || !process.env.WORDLES_REPO_NAME) {
+        return {
+            statusCode: 400,
+            body: JSON.stringify({
+                message: "Invalid HTTP request method or missing field values or missing environment variables."
+            })
+        };
+    }
+
+    const checkUrlExists = async function (wordleUrl) {
+        return new Promise(function (resolve, reject) {
+            const req = http.request({
```

Comment:
It seems quite possible this could be an HTTPS request in some cases and so might be worth drafting in axios to abstract over this - also it will save you doing the work parsing the URL into fields